src/tokens/saga.ts

```diff
@@ -39,56 +38,19 @@ export interface FetchedTokenBalance {
   balance: string
 }
 
-interface UserBalancesResponse {
-  userBalances: {
-    balances: FetchedTokenBalance[]
-  }
-}
-
 export async function fetchTokenBalancesForAddress(
   address: string
 ): Promise<FetchedTokenBalance[]> {
-  const chainsToFetch = getSupportedNetworkIdsForTokenBalances()
-  const userBalances = await Promise.all(
-    chainsToFetch.map(async (networkId) => {
-      const response = await fetch(`${networkConfig.blockchainApiUrl}/graphql`, {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-          Accept: 'application/json',
-        },
-        body: JSON.stringify({
-          query: gql`
-            query FetchUserBalances($address: Address!, $networkId: NetworkId) {
-              userBalances(address: $address, networkId: $networkId) {
-                balances {
-                  tokenId
-                  tokenAddress
-                  balance
-                }
-              }
-            }
-          `,
-          variables: {
-            address,
-            networkId: networkId.replaceAll('-', '_'), // GraphQL does not support hyphens in enum values
-          },
-        }),
-      })
+  const networkIds = getSupportedNetworkIdsForTokenBalances().join('&networkIds[]=')
+  const url = `${networkConfig.getWalletBalancesUrl}?address=${address}&networkIds[]=${networkIds}`
+  const response = await fetchWithTimeout(url)
```

Comment:
nit: a bit safer if the initial url already has search params, and ensuring all params are properly url encoded if needed
```suggestion
  const networkIds = getSupportedNetworkIdsForTokenBalances();
  const url = new URL(networkConfig.getWalletBalancesUrl); 
  url.searchParams.set('address', address);
  networkIds.forEach(id => url.searchParams.append('networkIds[]', id));
  const response = await fetchWithTimeout(url.toString());
```